crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1748,13 +1748,26 @@ impl<'src> Parser<'src> {
             .into();
         }
 
+        // We need to reset the context when parsing a parenthesized expression otherwise a
+        // parenthesized comparison expression will not be parsed in a `for` target.
+
+        // test_ok parenthesized_compare_expr_in_for
+        // for (x in y)[0] in iter: ...
+        // for (x in y).attr in iter: ...
+
+        // test_err parenthesized_compare_expr_in_for
+        // for (x in y)() in iter: ...
+        // for (x in y) in iter: ...
+        let current_context = ParserCtxFlags::empty();
```

Comment:
Nit: I think I would explicitly unset the `FOR_TARGET` flag to ensure the code doesn't reset other flags when we add new flags to the context.